python_modules/dagster-ext/dagster_ext/__init__.py

```diff
@@ -259,6 +277,33 @@ def _assert_param_json_serializable(value: _T, method: str, param: str) -> _T:
     return value
 
 
+_METADATA_VALUE_KEYS = frozenset(ExtMetadataValue.__annotations__.keys())
+
+
+def _normalize_param_metadata(
+    metadata: Mapping[str, Union[ExtMetadataRawValue, ExtMetadataValue]], method: str, param: str
+) -> Mapping[str, Union[ExtMetadataRawValue, ExtMetadataValue]]:
+    _assert_param_type(metadata, dict, method, param)
+    new_metadata: Dict[str, ExtMetadataValue] = {}
+    for key, value in metadata.items():
+        if not isinstance(key, str):
+            raise DagsterExtError(
+                f"Invalid type for parameter `{param}` of `{method}`. Expected a dict with string"
+                f" keys, got a key `{key}` of type `{type(key)}`."
+            )
+        elif isinstance(value, dict):
+            if not {*value.keys()} == _METADATA_VALUE_KEYS:
+                raise DagsterExtError(
+                    f"Invalid type for parameter `{param}` of `{method}`. Expected a dict with"
+                    " string keys and values that are either raw metadata values or dictionaries"
+                    f" with schema `{{raw_value: ..., type: ...}}`. Got a value `{value}`."
+                )
+            new_metadata[key] = cast(ExtMetadataValue, value)
+        else:
+            new_metadata[key] = {"raw_value": value, "type": None}
```

Comment:
type hardcoded to `None`?

Reply:
where is the related code on the orch side?

Reply:
Untyped metadata values always get run through this: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/definitions/metadata/__init__.py#L106-L135

Reply:
Ok. I would much rather have this be an explicit value rather than `None`. e.g. `infer`, `untyped` etc I think that would be more clear and obvious

Reply:
Good idea, I made it `"__infer__"`